app/src/main/kotlin/com/adesso/movee/scene/movie/MovieViewModel.kt

```diff
@@ -63,12 +59,8 @@ class MovieViewModel @Inject constructor(
     }
 
     private fun fetchPopularMovies() {
-        bgScope.launch {
-            val popularMoviesResult = fetchPopularMoviesUseCase.run(UseCase.None)
-
-            onUIThread {
-                popularMoviesResult.either(::handleFailure, ::postPopularMovieList)
-            }
+        fetchPopularMoviesUseCase.invoke(viewModelScope, UseCase.None) {
+            it.either(::handleFailure, ::postPopularMovieList)
```

Comment:
postPopularMovieList() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postPopularMovieList()